hw/ip_templates/otp_ctrl/dv/env/seq_lib/otp_ctrl_zeroization_vseq.sv.tpl

```diff
@@ -0,0 +1,271 @@
+      // Now randomize the partitions and address to zeroize
+      `DV_CHECK_RANDOMIZE_FATAL(this)
+
+      // First check if the partition zeroizable marker is set for the partition that contains the
+      // DAI address that was generated in the randomize method above.
+      // If the marker is not set, first set the marker before zeroizing the address location.
+      if (!part_is_zeroizable(part_idx)) begin
+        `uvm_info(`gfn, $sformatf("ZRing Non Zeroizable Partition: part_idx :%s, dai_addr :%x",
+                part_idx.name, dai_addr), UVM_LOW);
+        dai_zeroize(.addr(dai_addr));
+      end else begin
+        if (!zeroized_partition[part_idx]) begin
+          `uvm_info(`gfn, $sformatf("ZRing Partition: part_idx :%s, dai_addr :%x",
+                  part_idx.name, zeroized_offset(part_idx)), UVM_LOW);
+          dai_zeroize(.addr(zeroized_offset(part_idx)));
+          zeroized_partition[part_idx] = 1;
+          dai_rd(.addr(zeroized_offset(part_idx)), .rdata0(rdata0), .rdata1(rdata1),
+                 .skip_idle_check(1));
+        end
+
+        if (used_zeroized_addrs.exists(dai_addr)) begin
+          `uvm_info(`gfn, $sformatf("addr %0h is already written!", addr), UVM_MEDIUM)
+          continue;
+        end
+
+        if (set_dai_regwen && dai_regwen_state) begin
+          csr_rd(ral.direct_access_regwen, reg_rd_val);
+          `uvm_info(`gfn, $sformatf("Setting DAI Regwen to zero - Locking DAI"), UVM_LOW);
+          dai_regwen_state = 0;
+          csr_wr(ral.direct_access_regwen, dai_regwen_state);
+        end
+
+        `uvm_info(`gfn, $sformatf("ZRing(Pre): part_idx :%s, dai_addr :%x",
+                   part_idx.name, dai_addr), UVM_LOW);
+        dai_zeroize(.addr(dai_addr));
+        used_zeroized_addrs[dai_addr] = 1;
+
+        // Once Zeroization command completes, if zerorization is successful rdata0 should return the
+        // count of ones in the bits updated depending on the granule size, and rdata1 should
+        // return 0
+        csr_rd(ral.direct_access_rdata[0], rdata0);
+        if (is_granule_64(dai_addr))
+          csr_rd(ral.direct_access_rdata[1], rdata1);
+
+        `uvm_info(`gfn, $sformatf("ZRing(Post): part_idx :%s, dai_addr :%x, rdata0:%x, rdata1:%x",
+                  part_idx.name, dai_addr, rdata0, rdata1), UVM_LOW);
+      end
+    end
+
+    `uvm_info(`gfn, "Reseting OTP - Zeroization State should be saved", UVM_LOW);
+    do_otp_ctrl_init = 0;
+    dut_init();
+    dut_init_completed = 1;
+
+    // dut_init() also sets interrupt enables. For this sequence otp_operation_done is disabled as
+    // we are expliciting testing otp_error.
+
+    // TODO: However the prediction model for otp_operation_done
+    // does need a revisit to make sure the interrupt state registers are set at the time the RTL
+    // also sets the same register and not the current way the model is coded
+
+
+    if (used_zeroized_addrs.num() != 0) begin
```

Comment:
This uvm_info seems pretty lame, or just just useless. Perhaps it was useful during early development?